src/index.ts

```diff
@@ -31,6 +32,19 @@ const server: McpServer = new McpServer({
 
 registerTools(server);
 
+let clientName: string | undefined;
+
+function setClientName(name: string) {
+  clientName = name;
+}
+
+const origInitializeHandler =
```

Comment:
check if this works https://github.com/modelcontextprotocol/typescript-sdk/blob/621ccea997bf318ee99c7f64ce19609e838615fa/src/server/index.ts#L274